### DISCUSSION TITLE
Type-safe selectors

### DISCUSSION THREAD

**Discussion by Amareis** [vanilla-extract-css/vanilla-extract#107] [Ideas]
Currently, there is no any (compile-time or emit-time) checking of selector string correctness. Latest typescript can check it with template string types, but maybe it's better to create (or use existing, if any) some util to form selector string? And certainly string should be checked when css emitted.

**Comment by markdalgleish** [comment]
We currently validate that selectors in `style` blocks target `&`, so there's already a form of build-time validation there. Not sure what happens if you pass an invalid selector, though, but I'm assuming it fails. What's your experience been here? Would like to hear a bit more detail.

We don't do any validation on global selectors, though, so that's probably a good idea just to check that they're valid CSS. I wouldn't attempt to do this with TypeScript—doesn't sound very practical for us to write a selector parser out of template literal types. I'd leverage the selector parser we're already using.

**Comment by Amareis** [comment]
At least checking pseudo selectors on typos, may be? Full string validation is not possible ofc
